image_transport/include/image_transport/transport_hints.h

```diff
@@ -59,40 +63,20 @@ class TransportHints
    * Defaults to the local namespace.
    * @param parameter_name The name of the transport parameter
    */
-  // TODO(ros2) change to equivalent ROS2::TransportHints
-  TransportHints(const std::string& default_transport = "raw",
-                 /*const ros::TransportHints& ros_hints = ros::TransportHints(),*/
-                 /*const ros::NodeHandle& parameter_nh = ros::NodeHandle("~"),*/
+  TransportHints(const std::shared_ptr<rclcpp::SyncParametersClient>& parameter_client,
```

Comment:
How is this API supposed to be used?

Instead of requiring an asynchronous parameter query (including to wait for a future) I would rather just pass a `Node` to the constructor on which the synchronous `get_parameter()` method is invoked directly (at least as an alternative signature?).

After this has settled the docblock needs to be updated accordingly.